index.bs

```diff
@@ -1616,27 +1673,37 @@ partial interface MLGraphBuilder {
     **Returns:** an {{MLOperand}}. The permuted or transposed N-D tensor. 
 </div>
 
-## MLGraph ## {#api-mlgraph}
-The {{MLGraph}} interface represents a compiled computational graph. A compiled graph once constructed is immutable and cannot be subsequently changed.
+## MLTensor ## {#api-mltensor}
+The {{MLTensor}} interface represents a resource of a multidimensional array of numbers with a shape.
 
 <script type=idl>
-dictionary MLInput {
-  required (MLBufferView or WebGLTexture or GPUTexture) data;
-  sequence<long> dimensions;
-};
+typedef (MLBufferView or WebGLTexture or GPUTexture) MLResource;
 
-dictionary MLOutput {
-  (MLBufferView or WebGLTexture or GPUTexture) data;
-  sequence<long> dimensions;
+[SecureContext, Exposed=Window]
+interface MLTensor {
+  // Construct a tensor from a resource
+  constructor(MLResource resource, optional sequence<long> dimensions);
+
+  // The tensor dimensions
+  sequence<long> dimensions();
```

Comment:
should tensor also have dtype method

Reply:
+1 to add data type, probably reuse the [`MLOperandType`](https://webmachinelearning.github.io/webnn/#enumdef-mloperandtype)?

Reply:
Actually that wouldn't be needed. Remember that `MLTensor` is just a resource holder type. The data type of the resource is already specified when an operand is created either explicitly through the `MLOperandDescriptor` or implicitly through the execution of the relevant operation in the graph. 

The fact that there is an optional `dimensions` param in its ctor may have confused the matter, but it's only there to support the graph with free dimensions input where the input shape isn't known until the graph execution time. In a normal use case, it won't be needed, hence optional.

This conversation makes me realize that calling the type `MLTensor` is itself misleading because when you call something a tensor, you would naturally think that it has both a shape and a data type association. A more appropriate name is probably `MLResource` or `MLGraphResource`.